### ISSUE TITLE
同配置，本项目微调显存溢出，LLama factory 不溢出。

### ISSUE THREAD

**Issue by OKC13** [modelscope/ms-swift#7162]
如下，3卡。
<img width="654" height="498" alt="Image" src="https://github.com/user-attachments/assets/7e7d9bc2-ed30-49b0-8411-96605d572187" />

训练参数：
`CUDA_VISIBLE_DEVICES=1,2,3 NPROC_PER_NODE=3 MAX_PIXELS=451584 swift sft \
--model /llm/pretrain_models/Qwen/Qwen3-VL-4B-Instruct \
--dataset /llm/wjw/LLM_train/LLaMA-Factory/data/vlm_1216.json \
--train_type lora \
--lorap_lr_ratio 10 \
--freeze_vit false \
--freeze_aligner false \
--freeze_llm false \
--per_device_train_batch_size 1 \
--per_device_eval_batch_size 1 \
--split_dataset_ratio 0.05 \
--output_dir output_dir/ \
--num_train_epochs 3 \
--save_steps 100 \
--eval_steps 100 \
--save_total_limit 2 \
--logging_steps 10 \
--seed 42 \
--learning_rate 5e-5 \
--init_weights true \
--lora_rank 8 \
--lora_alpha 16 \
--adam_beta1 0.9 \
--adam_beta2 0.95 \
--adam_epsilon 1e-08 \
--weight_decay 0.1 \
--gradient_accumulation_steps 8 \
--max_grad_norm 1 \
--lr_scheduler_type cosine \
--warmup_ratio 0.05 \
--warmup_steps 0 \
--gradient_checkpointing false \
--report_to swanlab \
--deepspeed zero2`

## 以下是Llama factory 训练时参数
CUDA_VISIBLE_DEVICES=1,2,3 torchrun --nnodes 1 --nproc_per_node 3 --master_port=25267 src/train.py \
    --deepspeed /llm/wjw/LLaMA-Factory-main/ds_zero2_no_offload.json \
    --stage sft \
    --model_name_or_path /llm/pretrain_models/Qwen/Qwen3-VL-4B-Instruct\
    --do_train \
    --dataset vlm_1216 \
    --finetuning_type lora \
    --output_dir  checkpoints/Qwen3-VL-4B-Instruct-12-16 \
    --overwrite_cache \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 8 \
    --lr_scheduler_type cosine \
    --logging_steps 10 \
    --save_steps 300 \
    --learning_rate 5e-6 \
    --num_train_epochs 3.0 \
    --cutoff_len 4096 \
    --bf16 true\
    --use_dora false\
    --plot_loss \
    --lora_rank 8\
    --lora_alpha 16\
    --lora_dropout 0.1\
    --lora_target all\
    --template qwen3_vl_nothink\
    --flash_attn auto\
    --loraplus_lr_ratio 16.0\
    --preprocessing_num_workers 60 \
    --overwrite_output_dir\
    --val_size 0.05 \
    --per_device_eval_batch_size 1 \
    --eval_strategy steps \
    --eval_steps 50

**Comment by Jintao-Huang** [comment]
--gradient_checkpointing true

**Comment by Jintao-Huang** [comment]
--per_device_train_batch_size 8 --gradient_accumulation_steps 1 --padding_free true --attn_impl flash_attention_2

**Comment by OKC13** [comment]
非常感谢，已解决
